src/main/java/Duke.java

```diff
@@ -1,10 +1,431 @@
+import Tasks.*;
+import Exceptions.*;
+
+import java.util.Scanner;
+import java.util.Vector;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Objects;
+
 public class Duke {
-    public static void main(String[] args) {
+    protected static final int EVENT_KEYWORD_LENGTH = 6;
+    protected static final int DEADLINE_KEYWORD_LENGTH = 9;
+    private static final String FILE_DIRECTORY_LOCATION = Paths.get(".", "save").toString(); ////////
+    protected static final String DATA_FILE_NAME = "duke" + ".txt";
+    private static final String DATA_FILE_PATH = Paths.get(FILE_DIRECTORY_LOCATION, DATA_FILE_NAME).toString();//////////
+    //protected static final String DATA_FILE_PATH = "src/main/java/" + DATA_FILE_NAME;
+    protected static final String SEPARATOR = "|";
+    protected static final String TASK_DONE_FLAG = "1";
+    protected static final String TASK_NOT_DONE_FLAG = "0";
+    public static Vector<Task> tasks = new Vector<>(); // list of tasks
+
+    public static void createFile() { // create file if it is not present
+        File fileHandle = new File(DATA_FILE_PATH);
+        if (!fileHandle.exists()) {
+            try {
+                fileHandle.createNewFile();
+            } catch (IOException e) {
+                System.out.println(fileHandle.getParentFile().mkdirs()); ////////
+                System.out.println("Error - An error has occurred when creating 'duke.txt': ");
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    public static void fileParse(String filePath) {
+        File fileHandler = new File(DATA_FILE_PATH);
+        Scanner fileScanner = null; 
+        try {
+            fileScanner = new Scanner(fileHandler);
+            fileScanner.useDelimiter(System.lineSeparator());
+        } catch (FileNotFoundException e) {
+            System.out.println("Error - 'duke.txt' not found");
+        }
+
+        while (fileScanner.hasNext()) {
+            String nextTask = fileScanner.next();
+            String[] taskTypes = nextTask.split(SEPARATOR);
+            Boolean taskDone = null;
```

Comment:
Similarly, consider renaming this variable to `isDone` to adhere to the coding standard.